cloudbuild/nightly/nightly.sh

```diff
@@ -28,38 +28,98 @@ case $STEP in
     exit
     ;;
 
+  # Create the cluster
+  create_clusters_bounded_small_table)
+    #  Get the timestamp to append to cluster name.
+    timestamp=$(date +"%Y%m%d%H%M%S")
+    # 1. Create the first cluster for bounded read.
+    # - Modify the cluster name for all tests.
+    CLUSTER_NAME_SMALL_TEST="$CLUSTER_NAME_SMALL_TEST"-"$timestamp"
+    # - call script that creates cluster with retries.
+    source cloudbuild/nightly/scripts/create_dataproc_cluster.sh "$CLUSTER_NAME_SMALL_TEST" "$REGION_ARRAY_STRING_SMALL_TEST" "$NUM_WORKERS_SMALL_TEST" "$REGION_SMALL_TEST_FILE" "$WORKER_MACHINE_TYPE_SMALL_BOUNDED"
+    # - save the cluster for future uses
+    echo "$CLUSTER_NAME_SMALL_TEST" > "$CLUSTER_SMALL_TEST_FILE"
+    exit
+    ;;
```

Comment:
I see a template which can be defined as one function and called with relevant arguments. There is code repetition right now.

```
  create_cluster)
    timestamp=$(date +"%Y%m%d%H%M%S")
    CLUSTER_NAME="$CLUSTER_NAME"-"$timestamp"
    source cloudbuild/nightly/scripts/create_dataproc_cluster.sh "$CLUSTER_NAME" "$REGION_ARRAY_STRING" "$NUM_WORKERS" "$REGION_NAME_TEST_NAME_FILE" "$WORKER_MACHINE_TYPE"
    echo "$CLUSTER_NAME_TEST_NAME" > "$CLUSTER_NAME_TEST_NAME_FILE"
    exit
    ;;
```

Reply:
Program arguments ($FOO) can be converted to method arguments when invoking this function.